### PR TITLE
Backend MVP

### PR DESCRIPTION
**Pull request by notlfish**
# What?
Pull request to add a Minimum Viable Product for the Hotel Services backend API, according to the [specifications](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md).

For this MVP the following tasks were done:
- [x] Specify and document the requests and responses handled by the API, see [the documentation](https://github.com/RokoVarano/hotel_services_backend#api-documentation)
- [x] Design a database for persisting the data, see [the ERD](https://github.com/RokoVarano/hotel_services_backend/blob/develop/docs/ERD.png).
- [x] Add configuration for allowing CORS and routing the requests.
- [x] Add models and migrations for database entities.
- [x] Add models validations.
- [x] Add controllers.
- [x] Add tests.
- [x] Add seeds to easily interact with the frontend during development.

## Note for the next reviewer:
The backend of this project had no changes since the first review, but there are lots of improvements on the frontend. Please, review them.

### For more details on the implementation of each task, see the messages of the [closed Pull Requests](https://github.com/RokoVarano/hotel_services_backend/pulls?q=is%3Apr+is%3Aclosed) labeled as `MVP 1`.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby '3.0.2'
+
```
Comment (mcrd25) [inline Gemfile:5]: [OPTIONAL]
Swagger is good option for documentation, it'd make your API a bit more interactive. See [rswag gem](https://github.com/rswag/rswag)

File: README.md [hunk 2]
```diff
@@ -104,12 +113,12 @@ Body (example):
...
+    "errors": ["Service removal has failed"]
 }
 ```
 
 ## List a given user's reservations
-### Endpoint: GET api/v1/user/:id/reservations
+### Endpoint: GET api/v1/users/:id/reservations
 #### Response
 Successful:
 Code: 200
```
Comment (mcrd25) [inline README.md:124]: Good job with this documentation! as mentioned above, you can go a step further usering swagger)

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi Team, 

As expected, good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!



## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._
- The majority (if not all since I see no deal breaker here) of work needs to be done on the Front-end

You can also consider:
-  I don't really have much else to say, but maybe be more consistent with tests. If you test name individually in negative scenario you should test it individually in positive scenario.
- Check how your organize your code in general

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by mcrd25** [comment]
I believe I have reviewed this PR already. So I invalidated the review request since no changes were made. IT seems like you have some issues with creating services and recommendations, please double check this. If you need further assistance you are free to reach out to me personal and I'd be glad to trouble shoot with you

**Comment by notlfish** [comment]
> I believe I have reviewed this PR already. So I invalidated the review request since no changes were made. IT seems like you have some issues with creating services and recommendations, please double check this. If you need further assistance you are free to reach out to me personal and I'd be glad to trouble shoot with you

**Comment by notlfish** [comment]
## Note for the next reviewer:
The backend of this project had no changes since the first review, but there are lots of improvements on the frontend. Please, review them.

**Review by MuriloRoque — APPROVED** [review]
Hi @notlfish, @RokoVarano, @AdedayoOpeyemi, @afcastaneda223, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**